### PR TITLE
Fixes memory leak in SDF calculation - warp doesn't free memory

### LINKED ISSUE #2679: [Bug Report] Automate assembly Environment in SDF - Warp doesn't free memory of mesh points when scope is ended
SDF - Warp doesn't free memory of mesh points when scope is ended [bug].

### Describe the bug

Memory leak - reaches 100% of the system memory

### Steps to reproduce

Any example of Automate assembly that uses SDF. 
```
./isaaclab.sh -p source/isaaclab_tasks/isaaclab_tasks/direct/automate/run_w_id.py --assembly_id=ASSEMBLY_ID --train
```

### System Info

Describe the characteristic of your environment:

- Commit: [e.g. 8f3b9ca]
- Isaac Sim Version: 4.5.0-rc.36
- OS: Ubuntu 22.04
- GPU: A4500
- CUDA: 12.9
- GPU Driver: 575.57.08

### Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] I have checked that the issue is not in running Isaac Sim itself and is related to the repo

### Acceptance Criteria

Add the criteria for which this task is considered **done**. If not known at issue creation time, you can add this once the issue is assigned.

- [x] Memory leak is resolved

### PR DESCRIPTION
**Pull request by leondavi**
# Description

Thanks to Matteo Grimaldi, Saleh Nabi and Wonju Lee for identifying the source of this memory leak.

Add garbage collector calls since warp doesn't handle well the free memory.

Fixes #2679 

## Type of change

- Bug fix

## Screenshots

Before this change - the experiment hasn't been completed on 64GB RAM machines.
After  this change it requires less than 12GB RAM.

Screenshot 2025-06-12 at 18 31 38

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: source/isaaclab_tasks/isaaclab_tasks/direct/automate/industreal_algo_utils.py [hunk 1]
```diff
@@ -150,8 +153,14 @@ def get_sdf_reward(
...
         sdf_reward[i] = torch.mean(sdf_dist)
 
+        del mesh_copy
+        del mesh_points
+        del mesh_indices
+        del sampled_points
+
     sdf_reward = -torch.log(sdf_reward)
 
+    gp.collect() # Force garbage collection to free memory
```
Comment (Mayankm96) [inline source/isaaclab_tasks/isaaclab_tasks/direct/automate/industreal_algo_utils.py]: Should this be `gc.collect()`?

**Comment by kellyguo11** [comment]
@yijieg could you help review this one as well?

**Comment by yijieg** [comment]
Hi @kellyguo11 , this change looks good to me. It helps save memory but does not affect the reward we calculated, so it will not affect the learning performance.

File: source/isaaclab_tasks/isaaclab_tasks/direct/automate/industreal_algo_utils.py (hunk 1 above)
Comment (leondavi) [reply source/isaaclab_tasks/isaaclab_tasks/direct/automate/industreal_algo_utils.py]: Yes - I will fix this typo 
Thanks!

**Comment by kellyguo11** [comment]
@leondavi could you please run the formatter with `./isaaclab.sh -f`?
